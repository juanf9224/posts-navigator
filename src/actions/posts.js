
import { searchPosts } from '../services/post.service';

// SET_POSTS
export const setPosts = (posts) => ({
  type: 'SET_POSTS',
  posts
});

export const startSetPosts = () => {
  return (dispatch, getState) => {
    return searchPosts().then((res) => {
      dispatch(setPosts(res.data));
    });
  };
};

// EDIT_POST
export const editPostAction = (id, update) => ({
  type: 'EDIT_POST',
  id,
  update
});


