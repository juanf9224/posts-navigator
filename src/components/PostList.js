import React, { useContext } from 'react';

import Post from './Post';
import AutocompleteContext from "../context/autocomplete";

const PostsList = ({ listRef }) => {
  const {
    filteredPosts
  } = useContext(AutocompleteContext);

  const containsPosts = (filteredPosts && filteredPosts.length && filteredPosts);
  return (
    <ul ref={listRef}>
      {containsPosts
        ? filteredPosts.map((i) => (
            <Post
              key={i.id}
              post={i}
            />
          ))
        : null
      }
    </ul>
  );
};

export { PostsList as default };