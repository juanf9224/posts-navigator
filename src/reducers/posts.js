export const postsReducerDefaultState = {
  items: [],
  error: ''
};

export default (state = postsReducerDefaultState, action) => {
  switch (action.type) {
    case 'EDIT_POST':
      return {
        ...state,
        items: state.items.map((post) => {
          if (post.id === action.id) {
            return {
              ...post,
              ...action.update,
            };
          }
          return post;
        })
      };
    case 'GET_POSTS':
      return {
        ...state,
      };
    case 'GET_POSTS_SUCCESS':
      return {
        ...state,
        items: action.payload,
        error: '',
      };
    case 'GET_POSTS_ERROR':
      return { ...state, error: action.payload };
    default:
      return state;
  }
};