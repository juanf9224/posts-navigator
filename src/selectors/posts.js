export default (posts, { text, pagination }) => {
  const {
    pageNumber,
    itemsPerPage
  } = pagination;
  const start = (pageNumber - 1) * itemsPerPage;
  const end = start + itemsPerPage;
  
  if (!text) return { posts: posts.slice(start, end), matching: null };

  const filteredPosts = posts && posts
    .filter((p) => p.title.includes(text));
  return { posts: filteredPosts.length < start ? filteredPosts : filteredPosts.slice(start, end), matching: filteredPosts };
};
