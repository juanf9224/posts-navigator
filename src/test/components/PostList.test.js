import React from "react";
import { shallow } from "enzyme";

import PostList from "../../components/PostList";
import AutocompleteContext from "../../context/autocomplete";
import posts from '../fixtures/posts';

describe("PostList component test suite", () => {  
  let wrapper, listRef, filteredPosts;

  beforeEach(() => {
    listRef = {};
    filteredPosts = posts;
    wrapper = shallow(
      <AutocompleteContext.Provider value={{ filteredPosts }}>
        <PostList listRef={listRef}/>
      </AutocompleteContext.Provider>
    );
  });

  test("it should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
