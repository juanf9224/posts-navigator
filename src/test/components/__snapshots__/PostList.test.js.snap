// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList component test suite it should match snapshot 1`] = `ShallowWrapper {}`;
